### PR TITLE
Bjorncs/use sia with node admin

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

**Comment by bjorncs** [comment]
The SIA specific configuration in `config-server.def` is optional. We could make the configuration mandatory If there are use case for communicating with config servers without authentication.

**Comment by bjorncs** [comment]
PTAL

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/ConfigServerInfo.java [hunk 1]
```diff
@@ -86,6 +93,18 @@ public URI getConfigServerUri(String hostname) {
                 hostname -> URI.create(scheme + "://" + hostname + ":" + port)));
     }
 
+    private static ConfigServerConfig.Sia verifySiaConfig(ConfigServerConfig.Sia sia) {
+        List<String> configParams = Arrays.asList(
+                sia.credentialsPath(), sia.configserverIdentityName(), sia.hostIdentityName(), sia.trustStoreFile());
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/ConfigServerInfo.java]: the configserverIdentityName is actually not related SIA? I thought one of the conclusions from yesterday's discussion was that SIA provides exactly one identity (=hostIdentityName).

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java [hunk 2]
```diff
@@ -1,91 +1,86 @@
...
-            }
-
-            keyStoreRefresher.start();
-            return keyStoreRefresher;
-        });
+    SslConnectionSocketFactoryUpdater(SiaIdentityProvider siaIdentityProvider,
+                                      HostnameVerifier configServerHostnameVerifier) {
+        this.configServerHostnameVerifier = configServerHostnameVerifier;
+        this.sia = siaIdentityProvider;
+        if (siaIdentityProvider != null) {
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java]: Express this by making siaIdentityProvider Optional.

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java [hunk 3]
```diff
@@ -29,24 +33,31 @@
 
     private static final Duration REFRESH_INTERVAL = Duration.ofHours(1);
```
Comment (hakonhall) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java]: It's still 1 hour refresh interval?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java [hunk 4]
```diff
@@ -1,91 +1,86 @@
...
      */
-    public static SslConnectionSocketFactoryUpdater createAndRefreshKeyStoreIfNeeded(
-            ConfigServerInfo configServerInfo, String hostname) {
-        return new SslConnectionSocketFactoryUpdater(
-                configServerInfo,
-                hostname,
-                ConfigServerKeyStoreRefresher::new,
-                new SslConnectionSocketFactoryCreator());
+    public static SslConnectionSocketFactoryUpdater createAndRefreshKeyStoreIfNeeded(ConfigServerInfo configServerInfo) {
+        SiaIdentityProvider siaIdentityProvider = configServerInfo.getSiaConfig()
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java]: If you're unconditionally replacing the former implementation, you also need to fill the prerequisite sia config in ConfigServerConfig everywhere.

**Comment by hakonhall** [comment]
FYI: @freva

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/ConfigServerInfo.java (hunk 1 above)
Comment (bjorncs) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/ConfigServerInfo.java]: It's not related to SIA per se, but it is required for constructing a `HostnameVerifier` that accepts the certificate from configserver. I agree that it's somewhat misleading to have the parameter under `Sia`.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java (hunk 2 above)
Comment (bjorncs) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java]: I follow the official recommendation that `Optional` should only be used as return parameter.

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java (hunk 3 above)
Comment (bjorncs) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java]: Yes, the credentials are read from disk every hour, although `sia` daemon is typically configured to update once every day.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java (hunk 4 above)
Comment (bjorncs) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java]: See PR on the _hosted_ repository. Afaik, all places where that config is generated should populate the new parameters.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/ConfigServerInfo.java [hunk 5]
```diff
@@ -28,6 +29,7 @@
     private final Optional<KeyStoreOptions> keyStoreOptions;
     private final Optional<KeyStoreOptions> trustStoreOptions;
     private final Optional<AthenzIdentity> athenzIdentity;
+    private final ConfigServerConfig.Sia siaConfig;
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/component/ConfigServerInfo.java]: Will fix

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java (hunk 4 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java]: Sorry, for some reason I missed that PR.

**Review by hakonhall — APPROVED** [review]

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java (hunk 2 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java]: That seems counter-productive, let's discuss offline

**Pull request closed without merging**